### PR TITLE
added software firewall license plugin code

### PR DESCRIPTION
**Pull request by salsop**
## Description

Added code for the management of the Software Licensing Plugin, BootStrap Defiinittions, and License Managers.

## Motivation and Context

Currently, there is no support for the Software Licensing Plugin, and we need to add this to enable us to add this to the Terraform Provider.

## How Has This Been Tested?

I've ran tests using a local Go program to add, change, and delete these options.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

File: pnrm/plugins/swfwlicense/bootstrapdef/testdata_test.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+type tc struct {
+	desc   string
+	plugin plugin.Info
+	conf   Entry
+}
+
+func getTests() []tc {
+	return []tc{
+		{"basic project", plugin.Info{Name: "sw_fw_license", Installed: "yes", Version: "1.0.0"}, Entry{ Name: "bootstrapdef", Description: "some desc", Authcode: "abcdefg" } },
```
Comment (shinmog) [inline pnrm/plugins/swfwlicense/bootstrapdef/testdata_test.go:15]: Split the `Entry` params onto their own lines like the other testcase files.

File: pnrm/plugins/swfwlicense/licensemgr/entry.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+// entry_v1 data structure from XML
+type entry_v1 struct {
+	XMLName             xml.Name `xml:"entry"`
+	Name                string   `xml:"name,attr"`
+	Description         string   `xml:"description,omitempty"`
+	DeviceGroup         string   `xml:"device-group"`
+	TemplateStack       string   `xml:"template-stack"`
+	BootstrapDefinition string   `xml:"bootstrap-definition"`
+	AutoDeactivate		int		 `xml:"auto-deactivate"`
```
Comment (shinmog) [inline pnrm/plugins/swfwlicense/licensemgr/entry.go:78]: Looks like you haven't done `make fmt`.

File: pnrm/plugins/swfwlicense/licensemgr/pano.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	"github.com/PaloAltoNetworks/pango/util"
+)
+
+// Panorama is the client.Panorama namespace.
+type Panorama struct {
+	ns *namespace.Plugin
+}
+
+// GetBootstrapParams for specific licensemgr and returns a string of the bootstrap parameters and an error if needed
+func (c *Panorama) GetBootstrapParams(licensemgr string) (string, error) {
```
Comment (shinmog) [inline pnrm/plugins/swfwlicense/licensemgr/pano.go:15]: Rename this param to `manager` or something; this variable is named the same as this whole package.

File: pnrm/plugins/swfwlicense/licensemgr/pano.go [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+	if err != nil {
+		return "", err
+	} else {
+		err = xml.Unmarshal(body, &result)
+		if err != nil {
+			return "", err
+		} else {
+			return result.BootstarpParams, nil
+		}
+	}
```
Comment (shinmog) [inline pnrm/plugins/swfwlicense/licensemgr/pano.go:38]: Typically you just pass in the struct you want the answer unmarshalled into and let `Op()` do it for you.  If the result of this `op` command doesn't require special handling, then just pass in `&result` and let `Op()` do the work.

File: pnrm/plugins/swfwlicense/licensemgr/testdata_test.go [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+type tc struct {
+	desc   string
+	plugin plugin.Info
+	conf   Entry
+}
+
+func getTests() []tc {
+	return []tc{
+		{"basic project", plugin.Info{Name: "sw_fw_license", Installed: "yes", Version: "1.0.0"}, Entry{ Name: "licmanager", Description: "some desc", DeviceGroup: "dg", TemplateStack: "ts", BootstrapDefinition: "bsdef", AutoDeactivate: 5} },
```
Comment (shinmog) [inline pnrm/plugins/swfwlicense/licensemgr/testdata_test.go:15]: Separate the Entry params onto their own lines, please.

**Review by shinmog — CHANGES_REQUESTED** [review]
Nice work on this overall!

One additional thing:  please rename this package:

`pnrm/plugins/swfwlicense/licensemgr`

to this:

`pnrm/plugins/swfwlicense/manager`

The path as a whole already has "license" in it, so putting it again in the deeper path is unnecessary.

File: pnrm/plugins/swfwlicense/licensemgr/pano.go (hunk 3 above)
Comment (salsop) [reply pnrm/plugins/swfwlicense/licensemgr/pano.go:15]: I've left this as `licensemgr` as renamed the package to `manager`, so it differentiates between the two now.

File: pnrm/plugins/swfwlicense/bootstrapdef/testdata_test.go (hunk 1 above)
Comment (salsop) [reply pnrm/plugins/swfwlicense/bootstrapdef/testdata_test.go]: Sure, I've updated.

File: pnrm/plugins/swfwlicense/licensemgr/entry.go (hunk 2 above)
Comment (salsop) [reply pnrm/plugins/swfwlicense/licensemgr/entry.go]: Yep, sorry, forgot to run `go fmt`, but but done now :)

File: pnrm/plugins/swfwlicense/licensemgr/pano.go (hunk 4 above)
Comment (salsop) [reply pnrm/plugins/swfwlicense/licensemgr/pano.go]: Didn't realize the `op` command did, that so changed that too.

File: pnrm/plugins/swfwlicense/licensemgr/testdata_test.go (hunk 5 above)
Comment (salsop) [reply pnrm/plugins/swfwlicense/licensemgr/testdata_test.go]: done this too.

**Review by shinmog — APPROVED** [review]
lgtm

**Pull request merged**